compiler/modules/types/list.bal

```diff
@@ -572,13 +572,42 @@ function listSubtypeIntersect(Context cx, SubtypeData t1, SubtypeData t2) return
 }
 
 function listSubtypeDiff(Context cx, SubtypeData t1, SubtypeData t2) returns SubtypeData {
+    if isDisjoint(cx, <Bdd> t1, <Bdd> t2) {
+        return t1;
+    }
     Bdd b = memoSubtypeDiff(cx.listMemo, <Bdd>t1, <Bdd>t2);
     if b != false && cx.listMemo[b] == () {
         //io:println("fresh BDD: ", bddToString(b));
     }
     return b;
 }
 
+function isDisjoint(Context cx, Bdd t1, Bdd t2) returns boolean {
+    if t1 !is BddNode || t2 !is BddNode {
+        return false;
+    }
+    if t1.left is true && t1.middle is false && t1.right is false &&
```

Comment:
I would expect we would need to handle BDDs that are unions of positive atoms.

Reply:
fixed